examples/nemo/eg4/if_example.f90

```diff
@@ -0,0 +1,64 @@
+! -----------------------------------------------------------------------------
+!
+! Original code Governed by the CeCILL licence (http://www.cecill.info)
+!
+! -----------------------------------------------------------------------------
+! BSD 3-Clause License
+!
+! Code modifications Copyright (c) 2019, Science and Technology Facilities Council
+! All rights reserved.
+!
+! Redistribution and use in source and binary forms, with or without
+! modification, are permitted provided that the following conditions are met:
+!
+! * Redistributions of source code must retain the above copyright notice, this
+!   list of conditions and the following disclaimer.
+!
+! * Redistributions in binary form must reproduce the above copyright notice,
+!   this list of conditions and the following disclaimer in the documentation
+!   and/or other materials provided with the distribution.
+!
+! * Neither the name of the copyright holder nor the names of its
+!   contributors may be used to endorse or promote products derived from
+!   this software without specific prior written permission.
+!
+! THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+! AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+! IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+! DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+! FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+! DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+! SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+! CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+! OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+! OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+! -----------------------------------------------------------------------------
+! Author R. W. Ford, STFC Daresbury Lab
+
+! Code extracted from the tra_adv benchmark, with a SIGN intrinsic
+! replaced with equivalent code and the resultant expression
+! simplified. Temporarily made the scalar tmpx into an array as
+! scalars are not yet supported, see #472.
+program test_ifs
+
+  integer, parameter :: jpi=10, jpj=10, jpk=10
+  real, dimension(jpi,jpj,jpk) :: zwx, zslpx, tmpx
+  !real :: tmpx
+  integer :: ji, jj, jk
+  
+  DO jk = 1, jpk-1
+     DO jj = 2, jpj
+        DO ji = 2, jpi
+           !tmpx = zwx(ji,jj,jk) * zwx(ji-1,jj,jk)
```

Comment:
Is there a reason for keeping the original form of the code commented out? If not, I think it can be removed.

Reply:
Yes there is a reason, but possibly not one that you will support :-). They are to remind me of the original form of the example. The next thing I am working on is supporting scalars. Once scalars are supported then this example will be changed back to use a scalar. I mention that this current code is temporary and that it will use a scalar in the future and reference the associated issue in the comments at the start of the example. Is that enough or would you prefer me to remove them?

Reply:
No, that's enough - I'm feeling kind :-)